pkg/alerts/alertsHandler/cronJobHandler.go

```diff
@@ -116,6 +117,109 @@ func updateAlertStateAndCreateAlertHistory(alertDetails *alertutils.AlertDetails
 	return nil
 }
 
+// An Alert State can be updated to Firing, when the previous (IntervalCount - 1) Evaluations + Current Evaluation satisfies the conditions.
+// Meaning, the (IntervalCount - 1) evaluations and the current Evaluations State should be in either Pending or Firing.
+// The (IntervalCount - 1) evaluations will be Fetched from the AlertHistory Table.
+func shouldUpdateAlertStateToFiring(alertDetails *alertutils.AlertDetails, currentState alertutils.AlertState) bool {
+	if !alertutils.IsAlertStatePendingOrFiring(currentState) {
+		return false
+	}
+
+	intervalCount := alertDetails.EvalWindow / alertDetails.EvalInterval
+	if intervalCount == 0 {
+		log.Errorf("ALERTSERVICE: shouldUpdateAlertStateToFiring: EvalWindow=%v is less than EvalInterval=%v. Alert=%+v", alertDetails.EvalWindow, alertDetails.EvalInterval, alertDetails.AlertName)
+		return false
+	}
+
+	alertHistoryList, err := databaseObj.GetAlertHistoryByAlertID(&alertutils.AlertHistoryQueryParams{
+		AlertId:   alertDetails.AlertId,
+		Limit:     intervalCount - 1,
+		SortOrder: alertutils.DESC,
+	})
+	if err != nil {
+		log.Errorf("ALERTSERVICE: shouldUpdateAlertStateToFiring: Error getting AlertHistory. Alert=%+v & err=%+v.", alertDetails.AlertName, err)
+		return false
+	}
+
+	if len(alertHistoryList) < int(intervalCount-1) {
+		return false
+	}
+
+	for _, alertHistory := range alertHistoryList {
+		if !alertutils.IsAlertStatePendingOrFiring(alertHistory.AlertState) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func GetLatestAlertHistory(alertId string) (*alertutils.AlertHistoryDetails, error) {
+	alertHistoryList, err := databaseObj.GetAlertHistoryByAlertID(&alertutils.AlertHistoryQueryParams{
+		AlertId:   alertId,
+		Limit:     1,
+		SortOrder: alertutils.DESC,
+	})
+	if err != nil {
+		log.Errorf("ALERTSERVICE: GetLatestAlertHistory: Error getting AlertHistory. AlertId=%v & err=%+v.", alertId, err)
+		return nil, err
+	}
+
+	if len(alertHistoryList) == 0 {
+		return nil, nil
+	}
+
+	return alertHistoryList[0], nil
+}
+
+func handleAlertCondition(alertToEvaluate *alertutils.AlertDetails, isAlertConditionMatched bool, alertDataMessage string) error {
```

Comment:
The `AlertDetails` type has a `Message` field. Should you use that instead of passing this `alertDataMessage`?

Reply:
The `Message` in the `AlertDetails` struct is the Custom message that the user has set for this Alert. 
And the `alertDataMessage` contains the specific records or series that satisfied the Alert Condition.